pkgs/build-support/setup-hooks/auto-patchelf.sh

```diff
@@ -11,16 +11,13 @@ isExecutable() {
 }
 
 findElfs() {
-    find "$1" -type f -exec "$SHELL" -c '
-        while [ -n "$1" ]; do
-            mimeType="$(file -b -N --mime-type "$1")"
-            if [ "$mimeType" = application/x-executable \
-              -o "$mimeType" = application/x-sharedlib ]; then
-                echo "$1"
-            fi
-            shift
-        done
-    ' -- {} +
+    find "$1" -type f | while IFS='' read -r file; do
+        mimeType="$(file -b -N --mime-type "$file")"
+        if [ "$mimeType" = application/x-executable \
+            -o "${mimeType}" = application/x-sharedlib ]; then
```

Comment:
While we're at it, try running https://github.com/koalaman/shellcheck, it's in nixpkgs. I think it will complain about the `-o` being non-standard. It should also find other useful things.

Reply:
Fixed all issues in the file.